server/base/src/main/java/org/apache/accumulo/server/util/ratelimit/SharedRateLimiterFactory.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.server.util.ratelimit;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.WeakHashMap;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.LongAdder;
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.util.ratelimit.GuavaRateLimiter;
+import org.apache.accumulo.core.util.ratelimit.RateLimiter;
+import org.apache.accumulo.server.util.time.SimpleTimer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SharedRateLimiterFactory {
+  private static final long REPORT_RATE = 60000;
+  private static final long UPDATE_RATE = 1000;
+  private static SharedRateLimiterFactory instance = null;
+  private final Logger logger = LoggerFactory.getLogger(SharedRateLimiterFactory.class);
+  private final WeakHashMap<String,SharedRateLimiter> activeLimiters = new WeakHashMap<>();
+
+  private SharedRateLimiterFactory() {}
+
+  public static SharedRateLimiterFactory getInstance(SimpleTimer timer) {
+    synchronized (SharedRateLimiterFactory.class) {
+      if (instance == null) {
+        instance = new SharedRateLimiterFactory();
+
+        // Update periodically
+        timer.schedule(new Runnable() {
```

Comment:
Indeed, this is the entire purpose of calling the `update()` method on each rate limiter on a timer.  The only reason that `SharedRateLimiterFactory` even touches `AccumuloConfiguration` is so that it can grab hold of an instance of `SimpleTimer`.  Moving this polling to its own `Timer` object will make `SharedRateLimiterFactory` a self-contained singleton.